shortuuid.go

```diff
@@ -54,3 +56,14 @@ func NewWithAlphabet(abc string) string {
 func hasPrefixCaseInsensitive(s, prefix string) bool {
 	return len(s) >= len(prefix) && strings.EqualFold(s[:len(prefix)], prefix)
 }
+
+func hashedUUID(space uuid.UUID, data string) (u uuid.UUID) {
+	h := sha1.New()
+	h.Write(space[:])                                         //nolint:errcheck
+	h.Write(unsafe.Slice(unsafe.StringData(data), len(data))) //nolint:errcheck
```

Comment:
I noticed that in https://github.com/google/uuid/pull/181, you're not using `unsafe`?

Reply:
yes, in that PR signature accepts `data []byte`.
unsafe here is used to read input string as a byte slice without allocating additional objects.
So if that PR is merged and released, it could then be integrated here as:
`u = uuid.NewSHA1(uuid.NameSpaceDNS, unsafe.Slice(unsafe.StringData(data), len(data)))`
instead of
`u = uuid.NewSHA1(uuid.NameSpaceDNS, []byte(data))`
as this `[]byte` conversion sometimes adds an additional allocation.